miden-tx/src/tests/kernel_tests/test_faucet.rs

```diff
@@ -599,3 +616,18 @@ fn test_get_total_issuance_succeeds() {
 
     tx_context.execute_code(&code).unwrap();
 }
+
+pub fn assert_execution_errored(
+    execution_result: Result<Process<MockHost>, ExecutionError>,
+    expected_err_code: u32,
+) {
+    match execution_result {
+        Err(vm_processor::ExecutionError::FailedAssertion { clk: _, err_code, err_msg: _ }) => {
+            if err_code != expected_err_code {
+                panic!("Execution failed on assertion but was not expected error code")
+            }
+        },
+        Ok(_) => panic!("Execution was unexpectedly succesful"),
+        _ => panic!("Execution error was not as expected"),
+    }
+}
```

Comment:
This is really cool! A couple of comments:

- It would be great if we could convert it to a simple macro - e.g., `assert_execution_error!`. This way, if something goes wrong, source info will show to the file with the failed test rather than to here. (but if this is complicated, fine to keep as is).
- I would probably move this to a more central location rather than having it be under faucet tests.